code/game/objects/items/weapons/implants/implant_death_alarm.dm

```diff
@@ -68,4 +68,42 @@
 	if(..())
 		processing_objects.Remove(src)
 		return 1
-	return 0
\ No newline at end of file
+	return 0
+
+
+
+/obj/item/weapon/implant/death_alarm/dust
+	name = "duster implant"
+	desc = "An alarm which monitors host vital signs, transmitting a radio message and dusting the corpse on death."
+
+/obj/item/weapon/implant/death_alarm/dust/activate(var/cause)
```

Comment:
This whole thing is a huge amount of copy-pasta.

If you want to make it immune to EMPs, then make its `emp_act` return.

If you want it to dust on death, then add a var to the base alarm, that, if set, dusts the person on death (there's a proc for this too, by the by: `dust`) when it legitimately activates.

Then you won't have literally 90% of the death alarm code copy-pasted.